pkg/registry/common/dial/nse_client.go

```diff
@@ -94,19 +95,34 @@ func (c *dialNSEClient) Unregister(ctx context.Context, in *registry.NetworkServ
 		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
 	}
 
-	cc, _ := clientconn.Load(ctx)
+	cc, loaded := clientconn.LoadOrStore(ctx, newDialer(c.chainCtx, c.dialTimeout, c.dialOptions...))
 
 	di, ok := cc.(*dialer)
 	if !ok {
 		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
 	}
+
+	if loaded {
+		defer func() {
+			_ = di.Close()
+			clientconn.Delete(ctx)
+		}()
+		return next.NetworkServiceEndpointRegistryClient(ctx).Unregister(ctx, in, opts...)
+	}
+
+	err := di.Dial(ctx, clientURL)
+	if err != nil {
+		log.FromContext(ctx).Errorf("can not dial to %v, err %v. Deleting clientconn...", grpcutils.URLToTarget(clientURL), err)
+		clientconn.Delete(ctx)
+		return &emptypb.Empty{}, err
```

Comment:
I strongly believe all chain elements should pass `Close` call to next, regardless of errors, to clear any caches that could exist.
While `Unregister` is different, I still think it would be better to pass all calls to next, to clear caches and avoid potential memory leaks.

Reply:
We decided not to call next chain element on errors on internal meeting.